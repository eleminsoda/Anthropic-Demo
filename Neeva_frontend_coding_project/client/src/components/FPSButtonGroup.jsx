import React from "react";

import Button from "react-bootstrap/Button";
import ButtonGroup from "react-bootstrap/ButtonGroup";

export default function FPSButtonGroup({callback}){
    return (
        <ButtonGroup className="button-group" aria-label="FPS">
                <Button
                    variant="secondary"
                    onClick={() => {
                        callback(10);
                    }}
                >
                    10
                </Button>
                <Button
                    variant="secondary"
                    onClick={() => {
                        callback(30);
                    }}
                >
                    30
                </Button>
                <Button
                    variant="secondary"
                    onClick={() => {
                        callback(60);
                    }}
                >
                    60
                </Button>
            </ButtonGroup>
    )
}