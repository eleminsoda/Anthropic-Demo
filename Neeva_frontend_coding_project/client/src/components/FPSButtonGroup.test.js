import renderer from 'react-test-renderer';
import FPSButtonGroup from './FPSButtonGroup';

it('the button group should render correctly', () => {
  const component = renderer.create(
    <FPSButtonGroup setFPS={null} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});