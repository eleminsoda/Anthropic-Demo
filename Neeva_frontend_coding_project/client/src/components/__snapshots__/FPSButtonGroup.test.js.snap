// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button group should render correctly 1`] = `
<div
  aria-label="FPS"
  className="button-group btn-group"
  role="group"
>
  <button
    className="btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    10
  </button>
  <button
    className="btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    30
  </button>
  <button
    className="btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    60
  </button>
</div>
`;
